src/mesh/grids/straight_periodic_cube.cpp

```diff
@@ -24,24 +24,37 @@ void straight_periodic_cube(std::shared_ptr<TriangulationType> &grid,
 
     // Definition for each type of grid
     std::string grid_type_string;
-    if(dim==3) {
+    const bool colorize = true;
+    dealii::GridGenerator::hyper_cube(*grid, domain_left, domain_right, colorize);
+    if (dim == 1){
```

Comment:
use `if constexpr(dim == 1)` and similarly for the other ones since this is a template parameter

Reply:
Okay, fixing in the next commit.